tools/rewriter/SourceRewriter.cpp

```diff
@@ -1207,6 +1315,36 @@ int main(int argc, const char **argv) {
   }
   CompilationDatabase &comp_db = *MaybeCmds;
 
+  std::set<std::string> LibraryFilesSet;
+
+  // Read library and rewrite filenames from input files for library-only mode
+  if (LibraryOnlyMode) {
+    assert(SourceFiles.empty());
```

Comment:
Does the order between `LibraryFiles` and `RewriteFiles` matter? If not, you could get rid of the latter (per my comment above) and use `SourceFiles` in its place. You could copy its original contents to `RewriteFilesSet` first, then add in `LibraryFiles`.